### PR TITLE
EditableText fires onChange on escape

### LINKED ISSUE #687: EditableText not firing an onChange on esc
### Bug report

- __Package version(s)__: 1.9.0
- __Browser and OS versions__:  56.0.2924.87 (64-bit) Mac

#### Steps to reproduce

1. Start with your `EditableText` component empty
1. Click on it
1. Type "MyText"
1. Press `esc`

#### Actual behavior

1. onChange fires on each keystroke when typing "MyText"
1. onChange doesn't fire on `esc`
1. `EditableText` component goes out of focus and retains the text "MyText"

#### Expected behavior

1. onChange fires on each keystroke when typing "MyText"
1. onChange fires again on `esc`
1. `EditableText` component goes out of focus and goes back to being empty

This was fine in 1.8.0. Seems related to the various `EditableText` fixes that went into 1.9.0.

### PR DESCRIPTION
**Pull request by giladgray**
#### Fixes #687

### CODE REVIEW

File: packages/core/test/editable-text/editableTextTests.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ import * as ReactDOM from "react-dom";
 import * as Keys from "../../src/common/keys";
 import { EditableText } from "../../src/index";
 
-describe("<EditableText>", () => {
+describe.only("<EditableText>", () => {
```
Comment (cmslewis) [inline packages/core/test/editable-text/editableTextTests.tsx]: Ooooops. Remove the `.only`.

File: packages/core/test/editable-text/editableTextTests.tsx [hunk 2]
```diff
@@ -58,6 +58,22 @@ describe("<EditableText>", () => {
             assert.deepEqual(changeSpy.args, [["hello"], [" "], ["world"]]);
         });
 
+        it("calls onChange when escape key pressed and value is reverted", () => {
```
Comment (cmslewis) [inline packages/core/test/editable-text/editableTextTests.tsx]: IMO this phrasing doesn't parse clearly. Would prefer:

```
calls onChange and reverts value when escape key is pressed
```

File: packages/core/test/editable-text/editableTextTests.tsx (hunk 1 above)
Comment (giladgray) [inline packages/core/test/editable-text/editableTextTests.tsx]: DANG!

File: packages/core/test/editable-text/editableTextTests.tsx (hunk 2 above)
Comment (giladgray) [inline packages/core/test/editable-text/editableTextTests.tsx]: your EDIT is exactly what i'm testing though: that onChange is called when escape key is pressed _and_ the value should be reverted to the last confirmed value.

File: packages/core/test/editable-text/editableTextTests.tsx [hunk 3]
```diff
@@ -58,6 +58,22 @@ describe("<EditableText>", () => {
             assert.deepEqual(changeSpy.args, [["hello"], [" "], ["world"]]);
         });
 
+        it("calls onChange when escape key pressed and value is unconfirmed", () => {
```
Comment (cmslewis) [inline packages/core/test/editable-text/editableTextTests.tsx]: Ah, gotcha. This makes sense.

**Pull request merged**